.github/ci-scripts/bash/functions-for-build

```diff
@@ -496,50 +496,85 @@ ops_ui_build_and_push() {
-      return 
-    fi
-    xml_file="${jacoco_dir}/jacoco.xml"
-    branch_covered=$(xmllint --xpath 'sum(//counter[@type="BRANCH"]/@covered)' "${xml_file}")
-    branch_missed=$(xmllint --xpath 'sum(//counter[@type="BRANCH"]/@missed)' "${xml_file}")
-    branch_total=$((branch_covered + branch_missed))
-    branch_percentage=$(echo "scale=0; ${branch_covered} * 100 / ${branch_total}" | bc)
-    statement_covered=$(xmllint --xpath 'sum(//counter[@type="INSTRUCTION"]/@covered)' "${xml_file}")
-    statement_missed=$(xmllint --xpath 'sum(//counter[@type="INSTRUCTION"]/@missed)' "${xml_file}")
-    statement_total=$((statement_covered + statement_missed))
-    statement_percentage=$(echo "scale=0; ${statement_covered} * 100 / ${statement_total}" | bc)
-    export BRANCH_COVERAGE="${branch_covered} ${branch_missed} ${branch_total} ${branch_percentage}"
-    export STATEMENT_COVERAGE="${statement_covered} ${statement_missed} ${statement_total} ${statement_percentage}"
-    echo "BRANCH_COVERAGE=\"${BRANCH_COVERAGE}\""
-    echo "STATEMENT_COVERAGE=\"${STATEMENT_COVERAGE}\""
-    cd ${jacoco_dir}/..
-    zip -r coverage.zip jacoco/*
-    cd -
-    mv ${jacoco_dir}/../coverage.zip ./
+    parse_and_push_coverage_report_java "$@"
+  elif [[ "${language}" == "python" ]]; then
+    parse_and_push_coverage_report_python "$@"
+  else
+    echo "Code coverage currently not supported for ${language} based repositories"
+  fi
+}
+
+parse_coverage_report_python() {
+  if [ -f "pytest.ini" ]; then
+    echo "Running python Unit test and generating code coverage report"
+    run_python_unit_test "$@"
+    coverage_report=$(find . -type f -name "coverage.json")
+    coverage_html=$(dirname "${coverage_report}")/htmlcov/index.html
+    covered_branches=$(jq '.totals.covered_branches' ${coverage_report})
+    missing_branches=$(jq '.totals.missing_branches' ${coverage_report})
+    total_branches=$(jq '.totals.num_branches' ${coverage_report})
+    branch_coverage_percentage=$(echo "scale=2; ${covered_branches} / ${total_branches} * 100" | bc)
+    covered_statements=$(jq '.totals.covered_lines' ${coverage_report})
+    missing_statements=$(jq '.totals.missing_lines' ${coverage_report})
+    total_statements=$(jq '.totals.num_statements' ${coverage_report})
+    statement_coverage_percentage=$(jq '.totals.percent_covered' ${coverage_report} | awk '{print int($1)}')
+    export BRANCH_COVERAGE="${covered_branches} ${missing_branches} ${total_branches} ${branch_coverage_percentage%.*}"
+    export STATEMENT_COVERAGE="${covered_statements} ${missing_statements} ${total_statements} ${statement_coverage_percentage}"
     flag=1
-  else 
-    echo "Skipping jacoco code coverage for non-java repository."
+  else
+    echo "pytest.ini not present skipping unit test and code coverage report generation"
+  fi
+}
+
+parse_and_push_coverage_report_python() {
+  flag=0
+  parse_coverage_report_python
+  if [ ${flag} -eq 1 ]; then
+    echo "Pushing coverage report to S3"
```

Comment:
We are having issues showing html for python. So instead lets put both HTML and json files in a zip and upload it.